lib/utils/sdk.ts

```diff
@@ -50,6 +49,46 @@ export async function getApplication(
 	return sdk.models.application.get(nameOrSlugOrId, options);
 }
 
+/**
+ * Given an string representation of an application identifier,
+ * which could be one of:
+ *  - name (including numeric names)
+ *  - slug
+ *  - numerical id
+ *  disambiguate and return a properly typed identifier.
+ *
+ *  Attempts to minimise the number of API calls required.
+ * TODO: Remove this once support for numeric App IDs is removed.
+ */
+export async function getTypedApplicationIdentifier(
+	sdk: BalenaSDK,
+	nameOrSlugOrId: string,
+) {
```

Comment:
For legibility purposes, I find it helpful to add explicit return types, even where Typescript can infer them.
```
export async function getTypedApplicationIdentifier(
    ...
): string | number {
    ...
```